### PR TITLE
No mocha

### PR DESCRIPTION
**Pull request by ajm188**
Recommend walking commit by commit.

There are a few hacky things I had to do to workaround some of the differences between rspec-mocks and mocha, namely https://github.com/ajm188/factory_girl/commit/356c974e09af6dd92c73b98eafe09d9377d2c867 and https://github.com/ajm188/factory_girl/commit/153a7f796cec688d76acef849553c9dbf05aa60a

### CODE REVIEW

**Comment by ajm188** [comment]
Oh I can probably remove mocha from the Gemfile/gemspec, too 🚀

**Comment by ajm188** [comment]
Going to let hound run again, then fix all the violations in a single commit.

Then I'll nuke bourne and mocha from the gemspec.

File: spec/factory_girl/evaluator_class_definer_spec.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@
   end
 
   it "evaluates the block in the context of the evaluator" do
-    dependency_attribute = stub("dependency", name: :dependency, to_proc: -> { 1 })
+    dependency_attribute = double("dependency", name: :dependency, to_proc: -> { 1 })
```
Comment (ajm188) [inline spec/factory_girl/evaluator_class_definer_spec.rb]: Hmmm, this dupe assignment was already there. I'll fix the line-length, but I'm going to ignore the dupe assignment for now.

File: spec/factory_girl/attribute/dynamic_spec.rb [hunk 2]
```diff
@@ -29,7 +29,22 @@
...
     before do
-      subject.stubs(attribute_defined_on_attribute: result)
+      RSpec.configure do |config|
+        config.mock_with :rspec do |mocks|
+          mocks.verify_partial_doubles = false
+        end
+      end
+      allow(subject).to(
+        receive(:attribute_defined_on_attribute).and_return result,
+      )
```
Comment (ajm188) [inline spec/factory_girl/attribute/dynamic_spec.rb]: looks I managed to break hound 😛  (this is actually valid ruby and `rspec` runs it happily

**Comment by ajm188** [comment]
So the build seems to consistently broken on jruby-1.9mode for rails 3.2. This seems consistent across many other PRs/branches, so it's likely unrelated to my changes.

I've fixed all the hound violations, and so now I'll remove mocha/bourne from the gemspec

**Comment by odlp** [comment]
Thanks for the contribution @ajm188.

I've added one commit to avoid toggling `#verify_partial_doubles` & rebased on master to get the updated Travis config which no longer has `jruby-1.9mode` with Rails 3.2.

Hopefully once this build is green I can merge this in!

**Comment by ajm188** [comment]
@odlp cool, thanks! Let me know if there's anything you need from me here :)

**Pull request merged**